### ISSUE TITLE
Sporthd's "new view" fails to launch on kodi 18

### ISSUE THREAD

**Issue by pitsi** [bugatsinho/bugatsinho.github.io#98]
So, I got the upgrade to 0.1.12 earlier on my libreelec 9.2.6 x64 system and tried sporthd again, from the "new view" entry as mentioned at the top. The menu does not open and this comes up in the log
```
2022-02-03 20:01:30.725 T:139824246167296   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeEncodeError'>
                                            Error Contents: 'ascii' codec can't encode characters in position 3965-3967: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.sporthdme/default.py", line 504, in <module>
                                                get_new_events(url)
                                              File "/storage/.kodi/addons/plugin.video.sporthdme/default.py", line 209, in get_new_events
                                                client.parseDOM(str(data), 'div', attrs={'class': "panel"})))
                                            UnicodeEncodeError: 'ascii' codec can't encode characters in position 3965-3967: ordinal not in range(128)
                                            -->End of Python script error report<--
2022-02-03 20:01:30.824 T:139824254560000   ERROR: GetDirectory - Error getting plugin://plugin.video.sporthdme/?description&iconimage=%2fstorage%2f.kodi%2faddons%2fplugin.video.sporthdme%2ficon.png&mode=15&name=%5bB%5d%5bCOLOR%20white%5dNEW%20VIEW%5b%2fCOLOR%5d%5b%2fB%5d&url=https%3a%2f%2f1.vecdn.pw%2fprogram.php
2022-02-03 20:01:30.833 T:139824419211392   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.sporthdme/?description&iconimage=%2fstorage%2f.kodi%2faddons%2fplugin.video.sporthdme%2ficon.png&mode=15&name=%5bB%5d%5bCOLOR%20white%5dNEW%20VIEW%5b%2fCOLOR%5d%5b%2fB%5d&url=https%3a%2f%2f1.vecdn.pw%2fprogram.php) failed
```
If it helps, kodi's locale on that system is set to greek, but the same happens when I set it to english. However, it works as it should on libreelec 10.0.1 which comes with kodi 19. Reinstalling does not help. Removing all its settings (= the entire `/storage/.kodi/userdata/addon_data/plugin.video.sporthdme/` folder) does not help too.

**Comment by bugatsinho** [comment]
thank you. maybe the name of an event have a special character and produce this error. i will look on that.
```UnicodeEncodeError: 'ascii' codec can't encode characters```

**Comment by bugatsinho** [comment]
I tried all day yesterday and i didn't get a single error in my device with Kodi18. Are you still getiing the same error? if so then you have to locate default.py line 38-39 and remove the (#)
then restart kodi. let me know

**Comment by pitsi** [comment]
Unfortunately yes. I tried it just now, with no links with special characters
```
2022-02-04 15:46:53.674 T:139892178274048   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeEncodeError'>
                                            Error Contents: 'ascii' codec can't encode characters in position 3965-3967: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.sporthdme/default.py", line 504, in <module>
                                                get_new_events(url)
                                              File "/storage/.kodi/addons/plugin.video.sporthdme/default.py", line 209, in get_new_events
                                                client.parseDOM(str(data), 'div', attrs={'class': "panel"})))
                                            UnicodeEncodeError: 'ascii' codec can't encode characters in position 3965-3967: ordinal not in range(128)
                                            -->End of Python script error report<--
2022-02-04 15:46:53.783 T:139892118419200   ERROR: GetDirectory - Error getting plugin://plugin.video.sporthdme/?url=https%3A%2F%2F1.vecdn.pw%2Fprogram.php&mode=15&name=%5BB%5D%5BCOLOR+white%5DNEW+VIEW%5B%2FCOLOR%5D%5B%2FB%5D&iconimage=%2Fstorage%2F.kodi%2Faddons%2Fplugin.video.sporthdme%2Ficon.png&description=

```
Uncommenting lines 38 and 39, as you suggested, did the trick and the addon works like before. Proof of the uncommented lines inside default.py
```
# sed -n 38,39p .kodi/addons/plugin.video.sporthdme/default.py
reload(sys)
sys.setdefaultencoding("utf-8")

```

**Comment by bugatsinho** [comment]
so it was in your end as it wasn't set up utf-8 encoding for your python

**Comment by pitsi** [comment]
Well, the kodi 18 (= python2) installations I have are on libreelec 9.2.x (x64 and rpi3b+) and coreelec 9.2.x (s905 tvbox) and they use pyhon 2.7.16. The arm based systems can not be upgraded further to libreelec 10.x or coreelec 19.x.
```
# python --version
Python 2.7.16
```
On the other hand, kodi 19 on libreelec 10 (x64) uses python 3.8.9
```
# python --version
Python 3.8.9
```
And kodi 19 on my pc uses python 3.9.x, which makes some addons, including sporthd, completely crash it 
```
2022-02-04 19:49:38.687 T:4983     INFO <general>: CPythonInvoker(8, /home/user/.kodi/addons/plugin.video.sporthdme/default.py): script successfully run
2022-02-04 19:49:38.753 T:4983     INFO <general>: Python interpreter stopped
```
But I am now used to it.

So, as you can see, there is no viable solution to any of the above. Will uncommenting the forementioned lines break anything when it comes to python3?

**Comment by bugatsinho** [comment]
FIXED WITH LAST UPDATE 0.1.13
